ssl/quic/quic_lcidm.c

```diff
@@ -63,20 +67,104 @@ struct quic_lcidm_st {
 #endif
 };
 
-static unsigned long bin_hash(const unsigned char *buf, size_t buf_len)
+/*
+ * SipHash implementation, adapted from
+ * https://github.com/veorq/SipHash
+ */
+#define ROTL(x, b) (uint32_t)(((x) << (b)) | ((x) >> (32 - (b))))
+#define SIPROUND \
+    do { \
+        v0 += v1;                                                              \
+        v1 = ROTL(v1, 5);                                                      \
+        v1 ^= v0;                                                              \
+        v0 = ROTL(v0, 16);                                                     \
+        v2 += v3;                                                              \
+        v3 = ROTL(v3, 8);                                                      \
+        v3 ^= v2;                                                              \
+        v0 += v3;                                                              \
+        v3 = ROTL(v3, 7);                                                      \
+        v3 ^= v0;                                                              \
+        v2 += v1;                                                              \
+        v1 = ROTL(v1, 13);                                                     \
+        v1 ^= v2;                                                              \
+        v2 = ROTL(v2, 16);                                                     \
+    } while (0)
+
+static uint64_t siphash(const unsigned char *buf, size_t buf_len, uint64_t key)
```

Comment:
We already have siphash implementation! So why wouldn't we use it? Of course then we would have to make enable-quic dependent on enable-siphash. Or we would have to change the siphash code to be always built and disable it just in providers.